extension/observer/endpointswatcher.go

```diff
@@ -114,3 +178,17 @@ type EndpointsLister interface {
 	// implemented by an observer looking for endpoints.
 	ListEndpoints() []Endpoint
 }
+
+func (ew *EndpointsWatcher) logEndpointEvent(msg string, notify Notify, endpoints []Endpoint) {
+	if ce := ew.logger.Check(zap.DebugLevel, msg); ce != nil {
+		fields := []zap.Field{zap.Any("notify", notify.ID())}
+		for _, endpoint := range endpoints {
+			if env, err := endpoint.Env(); err == nil {
+				if marshaled, e := json.Marshal(env); e == nil {
+					fields = append(fields, zap.String(string(endpoint.ID), string(marshaled)))
+				}
+			}
+		}
+		ce.Write(fields...)
```

Comment:
This should have a formatted string here to make searching the logs easier.

Reply:
Sorry I'm not following. Do you have an example of what you're referring to? json env objects by endpoint id seems informative to me.

Reply:
I realized you may be referring to the log message, and `logEndpointEvent` takes this as `msg` which is `"(added|removed|changed) endpoints"` depending on the caller.

https://pkg.go.dev/go.uber.org/zap#example-Logger.Check

Reply:
Yes, sorry.

I was thinking a long the lines of `ce.Write("Something has happened", fields...)` mainly so you can search for the `Something has happened`